### PR TITLE
feat(project): Add api to fetch project

### PR DESCRIPTION
**Pull request by gagantrivedi**

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -66,3 +66,98 @@ func (c *Client) UpdateFeatureState(featureState *FeatureState) (*FeatureState,
 	}
 	return &updatedFeatureState, nil
 }
+
+func (c *Client) GetProject(projectUUID string) (*Project, error) {
+	url := fmt.Sprintf("%s/projects/", c.baseURL)
```
Comment (matthewelwell) [inline client.go:71]: Should we add a get-by-uuid endpoint for projects?

File: client.go [hunk 2]
```diff
@@ -66,3 +66,98 @@ func (c *Client) UpdateFeatureState(featureState *FeatureState) (*FeatureState,
...
+	project := result[0]
+	return project, nil
+
+}
+
+func (c *Client) GetFeature(featureUUID string) (*Feature, error) {
+	url := fmt.Sprintf("%s/features/get-by-uuid/%s/", c.baseURL, featureUUID)
+	feature := Feature{}
+	resp, err := c.client.R().
+		SetQueryParams(map[string]string{
```
Comment (matthewelwell) [inline client.go:95]: I don't think we need this query param since we're including it in the URL now?

File: client.go [hunk 3]
```diff
@@ -66,3 +66,98 @@ func (c *Client) UpdateFeatureState(featureState *FeatureState) (*FeatureState,
...
+
+	if !resp.IsSuccess() {
+		return nil, fmt.Errorf("flagsmithapi: Error getting feature: %s", resp)
+	}
+	return &feature, nil
+}
+
+func (c *Client) CreateFeature(feature *Feature) error {
+	projectID := feature.ProjectID
+	if projectID == nil {
```
Comment (matthewelwell) [inline client.go:112]: in which scenarios do we get a feature with a ProjectUUID but not a ProjectID?

File: client.go [hunk 4]
```diff
@@ -66,3 +66,98 @@ func (c *Client) UpdateFeatureState(featureState *FeatureState) (*FeatureState,
...
+	}
+
+	if !resp.IsSuccess() {
+		return fmt.Errorf("flagsmithapi: Error creating feature: %s", resp)
+	}
+
+	return nil
+}
+
+func (c *Client) DeleteFeature(projectID, featureID int64) error {
```
Comment (matthewelwell) [inline client.go:135]: General Golang question... how come you have to specify that the featureID is `int64` but not the projectID?

Comment (matthewelwell) [inline client.go:135]: General Golang question... how come you have to specify that the featureID is `int64` but not the projectID?

File: client.go [hunk 5]
```diff
@@ -66,3 +66,98 @@ func (c *Client) UpdateFeatureState(featureState *FeatureState) (*FeatureState,
...
+	}
+
+	if !resp.IsSuccess() {
+		return fmt.Errorf("flagsmithapi: Error deleting feature: %s", resp)
+	}
+	return nil
+}
+
+func (c *Client) UpdateFeature(feature *Feature) error {
+	url := fmt.Sprintf("%s/projects/%d/features/%d/", c.baseURL, *feature.ProjectID, *feature.ID)
```
Comment (matthewelwell) [inline client.go:151]: If we're checking above to see if projectID is nil, shouldn't we do the same here?

File: client.go (hunk 4 above)
Comment (gagantrivedi) [reply client.go:135]: Ah, it's because if the arguments have the same type you can write it only once at the end

File: client.go (hunk 5 above)
Comment (gagantrivedi) [reply client.go:151]: The hidden assumption here is that if the user is updating the feature, they must have already read it, hence they will have the Project.ID. but now I think it's better to check and fetch the UUID if it's not present

File: client.go (hunk 3 above)
Comment (matthewelwell) [reply client.go:112]: Gotcha, yep, makes sense.

File: client.go (hunk 4 above)
Comment (matthewelwell) [reply client.go:135]: Ah, interesting 👍

File: client.go (hunk 5 above)
Comment (matthewelwell) [reply client.go:145]: Did you decide against doing this in the end?

Comment (gagantrivedi) [reply client.go]: Ah yes, I lost it during stash/apply

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**